core/src/main/resources/hudson/PluginManager/_table.js

```diff
@@ -9,30 +9,42 @@ function checkPluginsWithoutWarnings() {
 }
 
 Behaviour.specify("#filter-box", '_table', 0, function(e) {
-      function applyFilter() {
-          var filter = e.value.toLowerCase();
-          ["TR.plugin","TR.plugin-category"].each(function(clz) {
-            var encountered = {};
-            var items = document.getElementsBySelector(clz);
-            for (var i=0; i<items.length; i++) {
-                var visible = (filter=="" || items[i].innerHTML.toLowerCase().indexOf(filter)>=0);
-                var name = items[i].cells && items[i].cells.length > 1
-                        ? items[i].cells[1].getAttribute('data-id')
-                        : items[i].getAttribute("name");
-                if (visible && name != null) {
-                    if (encountered[name]) {
-                        visible = false;
-                    }
-                    encountered[name] = true;
+    function applyFilter() {
+        var filter = e.value.toLowerCase().trim();
+        var filterParts = filter.split(/ +/).filter (word => word.length > 0);
+        var items = document.getElementsBySelector("TR.plugin");
```

Comment:
`getElementsBySelector` is a prototype.js function, and should be avoided. The alternatives are using `querySelectorAll` (https://caniuse.com/#search=querySelectorAll). If the browser support is not good enough (especially due to Baidu-browser support, then I suggest using `getElementsByClassName('plugin')` and filter out the elements that are not `<tr>` within the function.

Reply:
Unchanged from before, just threw out the other, obsolete selector ☹️

Reply:
Looks like I didn't read the diff properly then 👍